configs/components/pxp-agent.rb

```diff
@@ -142,34 +142,36 @@
     pkg.directory File.join(settings[:logdir], 'pxp-agent')
   end
 
-  case platform.servicetype
-  when "systemd"
-    pkg.install_service "ext/systemd/pxp-agent.service", "ext/redhat/pxp-agent.sysconfig"
-    pkg.install_configfile "ext/systemd/pxp-agent.logrotate", "/etc/logrotate.d/pxp-agent"
-    if platform.is_deb?
-      pkg.add_postinstall_action ["install"], ["systemctl disable pxp-agent.service >/dev/null || :"]
-    end
-  when "sysv"
-    if platform.is_deb?
-      pkg.install_service "ext/debian/pxp-agent.init", "ext/debian/pxp-agent.default"
-      pkg.add_postinstall_action ["install"], ["update-rc.d pxp-agent disable > /dev/null || :"]
-    elsif platform.is_sles?
-      pkg.install_service "ext/suse/pxp-agent.init", "ext/redhat/pxp-agent.sysconfig"
-    elsif platform.is_rpm?
-      pkg.install_service "ext/redhat/pxp-agent.init", "ext/redhat/pxp-agent.sysconfig"
+  platform.get_service_types.each do |servicetype|
+    case servicetype
+    when "systemd"
+      pkg.install_service "ext/systemd/pxp-agent.service", "ext/redhat/pxp-agent.sysconfig", init_system: servicetype
+      pkg.install_configfile "ext/systemd/pxp-agent.logrotate", "/etc/logrotate.d/pxp-agent"
+      if platform.is_deb?
+        pkg.add_postinstall_action ["install"], ["if [ -d '/run/systemd/system' ] ; then systemctl disable pxp-agent.service >/dev/null || :; fi"]
```

Comment:
Why do we need this check if we're already in the "systemd" case?

Reply:
For debian platforms with both sysv and systemd you need to add all of the scripts, and then run the scripts based on what's currently on the system. You only know you're in the systemd section at build time, but this can change at runtime so it needs to be in the scripts.

Reply:
oookay! thanks!

Reply:
This broke automated deployment for us with Ubuntu, since `/run/systemd/system` does not exist in the installer environment, so services will not be activated anymore at all since this change.

Reply:
@olifre Is there a way in your environment to tell which init system it's running? We want to be able to support deb-ish systems running either sysv/upstart or systemd

Reply:
@underscorgan As outlined in the long-standing https://tickets.puppetlabs.com/browse/PA-136 , which is about the very same issue concerning runtime behaviour of the agent for the same very same failing test against the runtime directory, the correct / portable check appears to be to test whether `/sbin/init` links to `/lib/systemd/systemd`. This works in Ubuntu and Debian installer chroots, and I still have to see a systemd system where it fails ;-).

Reply:
Honestly, I don't remember why this was needed here. If this work is going to move forward I can investigate further, but I was working on this too long ago to remember edge cases I ran into while testing.